dev/DynamicDependency/MddDetourPackageGraph.cpp

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddDetourPackageGraph.h"
+
+#include "PackageGraphManager.h"
+
+#include <../Detours/detours.h>
+
+// Windows provides HRESULT_FROM_WIN32() but not the reverse. We need that for compat reasons.
+// Define WIN32_FROM_HRESULT() to covert an HRESULT to a Win32 Error Code if the HRESULT's facility
+// is FACILITY_WIN32. See https://devblogs.microsoft.com/oldnewthing/20061103-07/?p=29133 for more details
+#if !defined(WIN32_FROM_HRESULT)
+inline DWORD Win32FromHResult(const HRESULT hr)
+{
+    if (HRESULT_FACILITY(hr) == FACILITY_WIN32)
+    {
+        return static_cast<DWORD>(HRESULT_CODE(hr));
+    }
+    else
+    {
+        return static_cast<DWORD>(hr);
+    }
+}
+#define WIN32_FROM_HRESULT(hr) Win32FromHResult(hr)
+#endif
+
+static LONG (WINAPI * TrueGetCurrentPackageInfo)(
```

Comment:
WINAPI* 
(this is the Windows convention on top of the cppcore guidelines, not 100% sure it applies for Project Reunion but I am proposing that it does)